sirepo/sim_data/ml.py

```diff
@@ -62,3 +63,57 @@ def _lib_file_basenames(cls, data):
         if name:
             return [cls.lib_file_name_with_model_field("dataFile", "file", name)]
         return []
+
+
+class ArchiveManager:
+    import contextlib
+
+    _ARCHIVE_EXTENSIONS = (
+        ".tar.gz",
+        ".zip",
+    )
+
+    def __init__(self, file_path):
+        import zipfile
+        import tarfile
+
+        self.file_path = file_path
+        self.filename = file_path if isinstance(file_path, str) else file_path.basename
```

Comment:
`self.path = pkio.py_path(file_path)`

Then you can say self.path.filename where you need it. And, self.path.ext gives you a lot of what you want. If it is a gz file, then it is an archive, and you can further qualify.

Reply:
Aside: [.gz is not an archive](https://stackoverflow.com/questions/33592099/how-do-i-list-contents-of-a-gz-file-without-extracting-it-in-python), so we really do need to use .tar.gz explicitly. I played with `pathlib` a bit, but it got too complicated.

Reply:
[b5a6fd8](https://github.com/radiasoft/sirepo/pull/4736/commits/b5a6fd8c12d7927066d3b9bc60ad131a8daed3f6)